### PR TITLE
feat: add realtime price query

### PR DESCRIPTION
**Pull request by dolcalmi**
- Add realtimePrice query
  - Avoid issues with apollo client cache. 
  - Add support for wallet currencies (needed for the multi-currency support)
  - Add timestamp and id
- Add realtimePrice subscription

## Query
  ```graphql
  query {
    realtimePriceSAT: realtimePrice(walletCurrency: BTC, displayCurrency: "EUR"){
      id
      timestamp
      base
      offset
      walletCurrency
      displayCurrency
      currencyUnit
      formattedAmount
    }
    realtimePriceUSD: realtimePrice(walletCurrency: USD, displayCurrency: "EUR"){
      id
      timestamp
      base
      offset
      walletCurrency
      displayCurrency
      currencyUnit
      formattedAmount
    }
  }
  ```
## Subscription
  ```graphql
  subscription prices {
    realtimePrice(input: {walletCurrency: BTC, displayCurrency: "EUR"}) {
      errors {
        message
      }
      price {
        id
        timestamp
        base
        offset
        walletCurrency
        displayCurrency
        currencyUnit
        formattedAmount
      }
    }
  }
  ```

### CODE REVIEW

File: src/graphql/main/schema.graphql [hunk 1]
```diff
@@ -974,6 +975,14 @@ type Query {
     @deprecated(
       reason: "TODO: remove. we don't need a non authenticated version of this query. the users can only do the query while authenticated"
     )
+
+  """
+  Returns 1 Sat or 1 Cent price for the given quote display currency
+  """
+  realtimePrice(
```
Comment (nicolasburtey) [inline src/graphql/main/schema.graphql:982]: will also need the type on the subscription

Comment (dolcalmi) [reply src/graphql/main/schema.graphql:982]: Updated https://github.com/GaloyMoney/galoy/commit/087c47c24422414876cf60ae7ab55bf1e00bb51c

**Comment by bodymindarts** [comment]
From discussion on 01.02.2023:
```

realtimePrice(denominatorCurrency)

RealTimePrice {
  timestamp
  denominatorCurrency
  usdPrice: PriceOfOneUsdCent {
    unit
    offset
    amount
  }
  btcPrice: PriceOfOneSat {
    unit
    offset
    amount
  }
}

PriceOfOneSat implements AnyPrice {
  currencyUnit
  offset
  base
}
PriceOfOneUsdCent implements AnyPrice
  currencyUnit
  offset
  base
}

interface AnyPrice {
  currencyUnit
  offset
  base
}
```

```
interface IPriceService {
  listCurrencies(): Promise<PriceCurrency[] | PriceServiceError>
  getUsdRealTimePrice(
    args: GetRealTimePriceArgs,
  ): Promise<DisplayCurrencyPerUsdCent | PriceServiceError>
  getBtcRealTimePrice(
    args: GetRealTimePriceArgs,
  ): Promise<DisplayCurrencyPerSat | PriceServiceError>
  listHistory(args: ListHistoryArgs): Promise<Tick[] | PriceServiceError>
}
```

**Comment by dolcalmi** [comment]
closed in favor of:
- https://github.com/GaloyMoney/galoy/pull/2291
- https://github.com/GaloyMoney/galoy/pull/2292
- https://github.com/GaloyMoney/galoy/pull/2298

**Pull request closed without merging**